### PR TITLE
API support for GroupProjectBinding resources [part 2]

### LINKED ISSUE #10003: GroupProjectBinding API endpoints

### PR DESCRIPTION
**Pull request by pkprzekwas**
**What does this PR do / Why do we need it**:
Add POST, DELETE and PATCH endpoints to interact with GroupProjectBinding resources.

**Does this PR close any issues?**
Fixes #10003 

**Does this PR introduce a user-facing change?**:
```release-note
Add REST API for interacting with GroupProjectBinding resources.
```

### CODE REVIEW

File: pkg/ee/group-project-binding/handler/handler.go [hunk 1]
```diff
@@ -136,3 +170,211 @@ func GetGroupProjectBinding(ctx context.Context, request interface{},
...
+// swagger:parameters patchGroupProjectBinding
+type patchGroupProjectBindingReq struct {
+	groupProjectBindingReq
+
+	// in: body
+	// required: true
+	Body struct {
+		Role string `json:"role"`
+	}
+}
```
Comment (lsviben) [inline pkg/ee/group-project-binding/handler/handler.go]: do we want to allow changing the Group?

File: pkg/handler/v2/group-project-binding/groupprojectbinding.go [hunk 2]
```diff
@@ -59,3 +59,66 @@ func GetGroupProjectBindingEndpoint(
...
+			req,
+			userInfoGetter,
+			projectProvider,
+			privilegedProjectProvider,
+			bindingProvider,
+		); err != nil {
+			return nil, err
+		}
+		return nil, nil
+	}
```
Comment (lsviben) [inline pkg/handler/v2/group-project-binding/groupprojectbinding.go]: NIT:

maybe simpler would be:
```suggestion
		return nil, createGroupProjectBinding(
			ctx,
			req,
			userInfoGetter,
			projectProvider,
			privilegedProjectProvider,
			bindingProvider)
	}
```

File: pkg/ee/group-project-binding/handler/handler.go [hunk 3]
```diff
@@ -136,3 +170,211 @@ func GetGroupProjectBinding(ctx context.Context, request interface{},
...
+		},
+		Spec: kubermaticv1.GroupProjectBindingSpec{
+			ProjectID: req.ProjectID,
+			Group:     req.Body.Group,
+			Role:      req.Body.Role,
+		},
+	}); err != nil {
+		return common.KubernetesErrorToHTTPError(err)
+	}
+	return nil
```
Comment (lsviben) [inline pkg/ee/group-project-binding/handler/handler.go]: Usually in KKP we return the created/patched resources, could you do that as well to keep consistency?

File: pkg/ee/group-project-binding/handler/handler.go (hunk 1 above)
Comment (pkprzekwas) [reply pkg/ee/group-project-binding/handler/handler.go]: That was my initial thought. But I don't have strong opinion.

Comment (lsviben) [reply pkg/ee/group-project-binding/handler/handler.go]: I would allow, I dont see a reason why not

**Pull request merged**